dags/sustainment/jira_to_yaml.py

```diff
@@ -0,0 +1,290 @@
+                and fields[mapping["dataset_category"]]["value"] == "Yes"
+                else None,
+                "refresh_rate": fields[mapping["refresh_rate"]]["value"]
+                if fields[mapping["refresh_rate"]]
+                else None,
+                "owner_division": fields[mapping["owner_division"]][0]["value"]
+                if fields[mapping["owner_division"]]
+                else None,
+                "owner_section": fields[mapping["owner_section"]]
+                if fields[mapping["owner_section"]]
+                else None,
+                "owner_unit": fields[mapping["owner_unit"]]
+                if fields[mapping["owner_unit"]]
+                else None,
+                "owner_email": fields[mapping["owner_email"]]
+                if fields[mapping["owner_email"]]
+                else None,
+                "civic_issues": fields[mapping["civic_issues"]][0]["value"]
+                if fields[mapping["civic_issues"]]
+                else None,
+                "topics": [topic["value"] for topic in fields[mapping["topics"]]]
+                if fields[mapping["topics"]]
+                else None,
+                "tags": [
+                    {"name": item, "vocabulary_id": None}
+                    for item in fields[mapping["tags"]]
+                ]
+                if fields[mapping["tags"]]
+                else None,
+                "information_url": fields[mapping["information_url"]]
+                if fields[mapping["information_url"]]
+                else None,
+                "excerpt": fields[mapping["excerpt"]]["content"][0]["content"][0][
+                    "text"
+                ]
+                if fields[mapping["excerpt"]]
+                else None,
+                "limitations": fields[mapping["limitations"]]["content"][0]["content"][
+                    0
+                ]["text"]
+                if fields[mapping["limitations"]]
+                else None,
+                "notes": fields["description"]["content"][0]["content"][0]["text"]
+                if fields["description"]
+                else None,
+                "resources": {},
+            }
+            print(issue_metadata)
+            package_id = fields["summary"].lower().replace(" ", "-")
+
+            return {package_id: issue_metadata}
+        else:
+            logging.info("This is a request for updating exisitng dataset.")
+
+    def write_issue_content_to_yaml(issue_content, filename):
+        """Receives a json input and writes it to a YAML file"""
+
+        logging.info(f"Generating yaml file: {filename}")
+        try:
+            with open(YAML_DIR_PATH / filename, "w") as file:
+                yaml.dump(issue_content, file, sort_keys=False)
+        except PermissionError:
+            message = "Note: yaml file already exist, please double check!"
+            raise Exception(message)
+        return {"filename": filename}
+
+    def close_ticket(jira_issue_id):
```

Comment:
Consider changing the name of this function to something else, since it does not close the ticket

Reply:
right!!